tracer/src/Datadog.Trace/Debugger/Snapshots/DebuggerSnapshotSerializer.cs

```diff
@@ -80,6 +80,17 @@ public static void SerializeStaticFields(Type declaringType, JsonTextWriter json
                     return true;
                 }
 
+                if (source is UnreachableLocal unreachable)
+                {
+                    jsonWriter.WritePropertyName(variableName);
```

Comment:
You should follow the way we insert the variable name [throughout `SerializeInternal`](https://github.com/DataDog/dd-trace-dotnet/blob/25ae4f59590a76d18ad37a6a55f069c13d7592d9/tracer/src/Datadog.Trace/Debugger/Snapshots/DebuggerSnapshotSerializer.cs#L67):
```
                    if (variableName != null)
                    {
                        jsonWriter.WritePropertyName(variableName);
                    }
```
maybe we should also revisit what's the impact of not having the variable name on the correctness of the structure of the snapshot?

Reply:
It's a local, if we don't have is name, we don't write it at all. (agree about the second point!)